android/app/src/main/assets/js/core.js

```diff
@@ -465,7 +465,10 @@ window.addEventListener('DOMContentLoaded', async () => {
     }
 
     if (reader.generalSettings.val.removeExtraParagraphSpacing) {
-      html = html.replace(/<\s*br[^>]*>/gi, '\n').replace(/\n{2,}/g, '\n\n');
+      html = html
+        .replace(/([\u200B-\u200D\uFEFF]|&nbsp;)(?<=<p>\1)/g, '')
```

Comment:
are you sure `(?<=` would working properly
![image](https://github.com/user-attachments/assets/c59ae184-fa7f-43e4-b0bc-d8cdfe6a803e)

Reply:
With customJS the lookbehind works fine. I am sure if the regex of that feature has different flavour